client/quantum_serverless/core/provider.py

```diff
@@ -332,16 +336,122 @@ def get_compute_resources(self) -> List[ComputeResource]:
         return resources
 
     def create_compute_resource(self, resource) -> int:
-        """Create compute resource for provider."""
-        raise NotImplementedError
+        """
+        Create compute resource for provider.
+
+        First, create a compute_template based on the defined ComputeResource.
+        If the compute_template already exists (which shouldn't be the case,
+        exit silently. Otherwise, create the template. This template is
+        ephemeral and will be deleted when no longer needed (either the cluster
+        was created or we failed to create it and thus need to start over).
+
+        Then use that template to create a KubeRay Cluster. If the cluster
+        already exists, we exit silently. Users are only able to configure the
+        amount of cpu/memory and the number of worker replicas. The full spec
+        can be found in the KubeRay API spec under the "protoCluster"
+        definition:
+        https://github.com/ray-project/kuberay/blob/master/proto/swagger/cluster.swagger.json
+        """
+        template_name = self.name + "-template"
+        api_root = f"{self.host}/apis/v1alpha2/namespaces/{self.namespace}"
```

Comment:
Maybe I'm wrong but how do you see to move `api_root` to the constructor to just not repeat it in each method? Maybe it has no sense because it can change depending on what the method does but just to mention.

Reply:
I initially was thinking the same thing, but iirc it didn't play nicely with something... let me try again

Reply:
it was pylint that was complaining: `quantum_serverless/core/provider.py:271:0: R0902: Too many instance attributes (8/7) (too-many-instance-attributes)`... think it should be fine to bump that value up by one

Reply:
If we need it and it has sense I'm fine too 😄